interceptors/retry/options.go

```diff
@@ -45,6 +48,15 @@ type BackoffFunc func(attempt uint) time.Duration
 // with the next iteration. The context can be used to extract request scoped metadata and context values.
 type BackoffFuncContext func(ctx context.Context, attempt uint) time.Duration
 
+// OnRetryCallback called in retry attempts flow, which can be used to add additiional logic when retry occurs
+//
+type OnRetryCallback func(attempt uint, err error)
+
+// OnRetryCallback called in retry attempts flow, which can be used to add additiional logic when retry occurs
+//
+// The context can be used to extract request scoped metadata and context values.
```

Comment:
After merging these, lets simplify this comment a bit too.

```suggestion
// OnRetryCallback is the type of function called when a retry occurs.
```